### DISCUSSION TITLE
How to use the cartesian trajectory controller?

### DISCUSSION THREAD

**Discussion by devvaibhav455** [ros-industrial/universal_robot#642] [Q&A]
Hi,

I have a UR5 robot but before I test my stuff on the real robot, I want to make sure everything works fine in gazebo simulation. I want the robot’s EE to follow a 6-DOF pose continuously. I have got a 6-DOF joystick which publishes poses continuously and I want the robot's EE to follow it. I can do so easily using the franka emika cartesian impedance controller which they have already provided as an example and that makes me think that such a thing is possible.

For UR5, after [searching](https://forum.universal-robots.com/t/hallo-i-have-a-question-about-new-feature-cartesian-trajectory-controller-how-can-i-start-this-action-server-and-where-is-controller-yaml-file-which-i-need-to-modiefy/14115) I found out that pose_based_cartesian_traj_controller is something which I should be looking at. The answer requests us to go to beta-testing branch which then redirects to the staging branch

    This beta-testing branch is deprecated. The changes from this branch now live on the [staging](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/tree/staging) branch.

I cloned the staging branch but unsure how to move forward. Could someone please provide step-by-step (if that's not possible, some guidance)  instructions to do that?

System details: Ubuntu 20.04 + ROS noetic on amd64 architecture
PS: I can switch to ROS melodic as well if that’s a hard-and-fast requirement.

Thanks in advance.

**Comment by gavanderhoorn** [comment]
I believe you've posted this on the wrong repository.

The repository in your quote is [UniversalRobots/Universal_Robots_ROS_Driver](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver). You're posting this on `ros-industrial/universal_robot`.

The controller you refer to is not hosted nor maintained here.

I'd suggest you repost your question on [UniversalRobots/Universal_Robots_ROS_Driver](https://github.com/UniversalRobots/Universal_Robots_ROS_Driver).

**Reply by devvaibhav455** [reply]
I see. Thanks for correcting me @gavanderhoorn . I have posted another issue on the correct repo [https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/issues/639](url)
But personally, I have found working with UR robots on ROS challenging whereas panda is easier. Anyways, thanks for your help!
